hircluster.c

```diff
@@ -2098,85 +2098,83 @@ static int __redisClusterGetReply(redisClusterContext *cc, int slot_num,
 
 static redisClusterNode *node_get_by_ask_error_reply(redisClusterContext *cc,
                                                      redisReply *reply) {
-    sds *part = NULL, *ip_port = NULL;
-    int part_len = 0, ip_port_len = 0, ret;
-    dictEntry *de;
     redisClusterNode *node = NULL;
-
-    if (cc == NULL || reply == NULL) {
-        return NULL;
-    }
-
-    if (cluster_reply_error_type(reply) != CLUSTER_ERR_ASK) {
```

Comment:
Removed string-parsing checks which are not for free.
This is already checked right before this function is called.